src/zmq/zmqpublishnotifier.cpp

```diff
@@ -156,6 +157,18 @@ bool CZMQPublishHashBlockNotifier::NotifyBlock(const CBlockIndex *pindex)
     return SendMessage(MSG_HASHBLOCK, data, 32);
 }
 
+bool CZMQPublishHashBlockHeaderNotifier::NotifyBlock(const CBlockIndex *pindex)
+{
+    uint256 hash = pindex->GetBlockHash();
+    uint32_t height = pindex->nHeight;
+    LogPrint(BCLog::ZMQ, "zmq: Publish hashblock %s\n", hash.GetHex());
+    char data[36];
+    for (unsigned int i = 0; i < 32; i++)
+        data[31 - i] = hash.begin()[i];
+    memcpy(data + 32, (char *)&height, 4);
```

Comment:
Copying an integer this way is architecture-dependent. Look at the functions in the endian.h file.